src/lightning/fabric/utilities/load.py

```diff
@@ -203,3 +203,21 @@ def _load_tensor(t: _NotYetLoadedTensor) -> Tensor:
         return t._load_tensor()
 
     return apply_to_collection(collection, dtype=_NotYetLoadedTensor, function=_load_tensor)
+
+
+def _take_state_and_load_stateful(
```

Comment:
nit: i find this name odd: what about `_load_state_into`  or `_combine_state_with` ...?

Reply:
wanted to express that the elements get removed from the source dict, so "take" was more expressive of that vs. "load".

Reply:
What about `move_state_into`?